src/server/handlers/launchpad.js

```diff
@@ -562,15 +571,46 @@ export const getGitBranch = async (snap, token) => {
   }
 };
 
+const getSnapRepoData = async(snap, token) => {
+  let gitBranch, snapcraftData;
+  try {
+    // first try to check defaut branch
+    gitBranch = await getGitBranch(snap, token);
+    // then get snapcraft.yaml data - there is no point trying to fetch this if fetching branch fails
+    snapcraftData = await getSnapcraftData(snap.git_repository_url, token);
```

Comment:
This means that we're doing the round-trips to GitHub consecutively rather than concurrently, which is likely to make it slower.  As far as I can see the point of this is mainly to ensure that errors from `getSnapcraftData` are ignored if `getGitBranch` has already failed, not so much the edge-case optimisation of performing fewer requests.  So how about something like this:

    let gitBranchError = null, snapcraftDataError = null;
    const [gitBranch, snapcraftData] = await Promise.all([
      getGitBranch(snap, req.session.token)
        .catch((error) => { gitBranchError = error; }),
      getSnapcraftData(snap.git_repository_url, req.session.token)
        .catch((error) => { snapcraftDataError = error; })
    ]);
    if (gitBranchError !== null) {
      ...
    } else if (snapcraftDataError !== null) {
      ...
    }

That's untested and maybe you can come up with a better formulation, but hopefully you get the idea?

Reply:
Yes, thanks for suggestion. I haven't thought about catching errors inside of `Promise.all`.